library/src/webapp/js/vanilla.js

```diff
@@ -0,0 +1,454 @@
+/**********************************************************************************
+ * $URL:
+ * $Id:
+ ***********************************************************************************
+ *
+ * Copyright (c) 2017 The Sakai Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+/* jshint esversion: 8 */
+
+/**
+ * Vanilla class - Contains helper functions for vanilla js functionality
+ */
+class Vanilla {
+  /**
+   * DOM ready
+   * @param {Function} callback - Function to execute when DOM is ready
+   */
+  static domReady(callback) {
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', callback);
+    } else {
+      callback();
+    }
+  }
+
+  /**
+   * Add class
+   * @param {Element} element - Target element
+   * @param {String} className - Class to add
+   */
+  static addClass(element, className) {
+    if (element && className) {
+      const classes = className.split(' ');
+      for (const cls of classes) {
+        if (cls) {
+          element.classList.add(cls);
+        }
+      }
+    }
+  }
```

Comment:
```suggestion
  static addClasses(el, classes) {
    if (!el || !classes) return;
    classes.split(" ").forEach(clazz => el.classList.add(clazz);
  }
```

Reply:
Generally though, you can go el.classList.add("green", "yellow", "blue") so I'm not sure how useful this method is.